src/game/Objects/Player.cpp

```diff
@@ -15696,8 +15696,8 @@ bool Player::IsAllowedToLoot(Creature const* creature)
 
 float Player::GetMaxLootDistance(Unit const* pUnit) const
 {
-    float distance = GetCombatReach() + 1.333333373069763f + pUnit->GetCombatReach();
```

Comment:
But this function is based on the client, it uses this combat and adds the combat reach of both. Why is the constant not used in your new code? Are you making the loot distance be different server side than it is client side?

This the client's check whether target is in loot distance, decompiled by IDA.
```
double v2; // st7
float v4; // [esp+8h] [ebp+8h]

v2 = *(float *)&this->UnitBase.UnitData->unitFields.UNIT_FIELD_COMBATREACH
 + 1.333333373069763
 + *(float *)&a2->UnitData->unitFields.UNIT_FIELD_COMBATREACH;
if ( v2 >= 5.0 )
v4 = v2;
else
v4 = 5.0;
return v4;
```

Reply:
The same function used by the client for loot distance is used for melee spells:
highest of the two: (Player combat reach + the melee reach constant + target combat reach) or ( 5 yards)

https://github.com/vmangos/core/blob/50f014e1ef9fb36c5ce0df8f1d8dd9c276ef5619/src/game/Objects/Unit.cpp#L10310C35-L10310C35

I made it call to that function to signify to people looking at the code in the future that the two ranges are the same and determined by client. 

I should change [BASE_MELEERANGE_OFFSET](https://github.com/vmangos/core/blob/551b8418f2225fe4c4e0339a595ab3731bf90a97/src/game/Objects/UnitDefines.h#L70) from 1.33f to the full value extracted from the client.